File: apps/mobile/app/src/main/java/com/subhajit/mulberry/navigation/AppRoute.kt
package com.subhajit.mulberry.navigation

sealed class AppRoute(val route: String) {
    data object Bootstrap : AppRoute("bootstrap")
    data object AuthLanding : AppRoute("auth_landing")
    data object OnboardingName : AppRoute("onboarding_name")
    data object OnboardingDetails : AppRoute("onboarding_details")
    data object OnboardingWallpaper : AppRoute("onboarding_wallpaper")
    data object PairingHub : AppRoute("pairing_hub")
    data object InviteCodeEntry : AppRoute("invite_code_entry")
    data object InviteAcceptance : AppRoute("invite_acceptance")
    data object CanvasHome : AppRoute("canvas_home")
    data object CanvasSurface : AppRoute("canvas_surface")
    data object LockScreenPlaceholder : AppRoute("lockscreen_placeholder")
    data object WallpaperCatalog : AppRoute("wallpaper_catalog")
    data object WallpaperHelp : AppRoute("wallpaper_help")
    data object PairingHelp : AppRoute("pairing_help")
    data object Streak : AppRoute("streak")
    data object Settings : AppRoute("settings")
}

File: apps/macos/Sources/AppShell/AppRoute.swift
import Foundation

public enum AppRoute: String, CaseIterable, Identifiable, Hashable, Sendable {
    case bootstrap
    case authLanding
    case onboardingName
    case onboardingDetails
    case onboardingOverlay
    case pairingHub
    case inviteCodeEntry
    case inviteAcceptance
    case canvasHome
    case canvasSurface
    case overlayStatus
    case overlayDisplay
    case overlayHelp
    case pairingHelp
    case streak
    case settings

    public var id: String { rawValue }

    var title: String {
        switch self {
        case .bootstrap: "Bootstrap"
        case .authLanding: "Sign In"
        case .onboardingName: "Name"
        case .onboardingDetails: "Profile"
        case .onboardingOverlay: "Overlay Setup"
        case .pairingHub: "Pairing"
        case .inviteCodeEntry: "Invite Code"
        case .inviteAcceptance: "Accept Invite"
        case .canvasHome: "Canvas"
        case .canvasSurface: "Canvas Surface"
        case .overlayStatus: "Overlay"
        case .overlayDisplay: "Display"
        case .overlayHelp: "Overlay Help"
        case .pairingHelp: "Pairing Help"
        case .streak: "Streak"
        case .settings: "Settings"
        }
    }

    static let sidebarRoutes: [AppRoute] = [
        .canvasHome,
        .overlayStatus,
        .pairingHub,
        .streak,
        .settings
    ]
}
